web/apps/serve.go

```diff
@@ -57,6 +59,27 @@ func onboarding(c echo.Context) bool {
 	return c.QueryParam("registerToken") != ""
 }
 
+// handleIntent will allow iframes from another app if the current app is
+// opened as an intent
+func handleIntent(c echo.Context, i *instance.Instance, slug, intentID string) {
+	intent := &intents.Intent{}
+	if err := couchdb.GetDoc(i, consts.Intents, intentID, intent); err != nil {
+		return
+	}
+	allowed := false
+	for _, service := range intent.Services {
+		if slug == service.Slug {
+			allowed = true
+		}
+	}
+	if !allowed {
+		return
+	}
+	from := i.SubDomain(intent.Client).Host
+	hdr := fmt.Sprintf("%s %s", middlewares.XFrameAllowFrom, from)
+	c.Response().Header().Set(echo.HeaderXFrameOptions, hdr)
+}
+
```

Comment:
To be sure that I understand clearly : this means that to open an intent, an app need to open first an iframe on itself, and in this iframe the X-frame-options will allow to add a second iframe to the real service ?